src/datastore/postgres-store.ts

```diff
@@ -820,9 +820,29 @@ export class PgDataStore
       UPDATE names
       SET canonical = $2
       WHERE index_block_hash = $1 AND canonical != $2
+      RETURNING name
       `,
       [indexBlockHash, canonical]
     );
+    nameResult.rows.forEach(async name => {
+      await client.query(
+        `
+        WITH latest_name AS (
+          SELECT id
+          FROM names
+          WHERE name = $1 AND canonical = true
+          ORDER BY registered_at DESC LIMIT 1
+        )
+        UPDATE names 
+        SET latest = false
+        WHERE name = $1 AND canonical =true
```

Comment:
The `latest_name` CTE looks good for finding the latest name, but the UPDATE expression doesn't seem right. 
Also, it looks like the multiple UPDATE expressions here are intentional -- multiple sql statements in a single query should never be used.

It looks like what you might want to do instead is add another CTE like `non_canonical_latest` that identifies row (should only be one) that is now non-canoncal but still latest, like `WHERE canonical = false AND latest = true`, then change the `UPDATE names SET latest = <...>` expression to be `true` if the row matches the `latest_name` CTE, and `false` if the row matches the `non_canonical_latest` CTE.